docs/source/download.rst

```diff
@@ -1,111 +1,200 @@
-see devito when doing imports like `from devito import Function`. Any code using
-Devito can be executed using this method.
+.. _Jupyter: https://jupyter.org/
 
-Issues? Ask us for help on Slack_. Also look at our list of known
-installation issues_.
+For those aiming to develop in Devito using Docker, you can use docker-compose
 
-.. _Jupyter: https://jupyter.org/
+.. code-block:: shell
+
+   # 1. Clone the code
+   git clone https://github.com/devitocodes/devito.git
+   cd devito
+
+   # 2. Start a bash shell with Devito
+   docker-compose run devito /bin/bash
+
+Docker for GPUs
+```````````````
+The GPU image differs from the CPU image in the additional compilers and toolkits necessary to run on GPUs. It contains the same installation and usage of Devito as the CPU image.
+
+Requirements:
+
+- Install the `NVIDIA container toolkit <https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#setting-up-nvidia-container-toolkit>`_.
+
+Available tags:
+
+- gpu-dev: for the latest development version (current GitHub master)
+- gpu-latest: for the latest GitHub release
+- gpu-vX.X.X: for a specific release (Release tags: https://github.com/devitocodes/devito/tags)
+
+.. code-block:: shell
+
+   # 1. Pull Devito image
+   docker pull devitocodes/devito:gpu-latest
+
+   # 2. (Optional but recommended) Test installation
+   docker run --gpus all --rm --name testrun 'devitocodes/devito:gpu-latest' pytest tests/test_gpu_openacc.py
+
+   # 3. Start a bash shell with Devito
+   docker run --gpus all --rm -it devitocodes/devito:gpu-latest /bin/bash
+
+For those aiming to develop in Devito using Docker, you can use docker-compose
+
+.. code-block:: shell
+
+   # 1. Clone the code
+   git clone https://github.com/devitocodes/devito.git
+   cd devito
+
+   # 2. Start a bash shell with Devito
+   docker-compose run devito.nvidia /bin/bash
+
+pip/conda installation
+----------------------
+
+User route
+``````````
 
-Pip installation
-----------------
 This is the recommended method when setting up Devito as part of a larger project
-that uses Devito among other python packages. To install Devito using `pip`, simply
-do:
+that uses Devito among other python packages. You can use Devito either in a `Python3 virtual environment`_ or in a `Conda environment`_.
+Devito is available as a `pip package`_ in PyPI. To install the latest Devito release along with any additional dependencies, follow:
+
+.. code-block:: shell
+
+   pip install devito
+   # Instead, to install additional dependencies
+   # pip install -e .[extras,mpi,nvidia]
 
+To install the latest Devito development version (current GitHub master) (without the tutorials), follow:
 
 .. code-block:: shell
 
-   pip install --user git+https://github.com/devitocodes/devito.git
+   pip install git+https://github.com/devitocodes/devito.git
+   # Instead, to install additional dependencies
+   # pip install git+https://github.com/devitocodes/devito.git#egg=project[extras,mpi,nvidia]
 
```

Comment:
~~~suggestion
Another option is to use the [Nix package manager](https://nixos.org/). To use it, first [install nix](https://nixos.org/download.html), and then execute the following command in the terminal:

`nix-shell -p python3Packages.devito`

If you want recipe that does that and allows you to share your development environment with others, you can use this file as an example:
```nix
# python.nix
with (import <nixpkgs> {});
let
  my-python-packages = python-packages: with python-packages; [
    devito
    # other python packages you want
  ];
  python-with-my-packages = python3.withPackages my-python-packages;
in
mkShell {
  buildInputs = [
    python-with-my-packages
  ];
}
```

and then execute:

`nix-shell python.nix`

If you need a more refined control over dependencies versions to ensure reproducibility, [this template](https://github.com/AtilaSaraiva/devito-nix-template) might be of use.
~~~